### DISCUSSION TITLE
How to get the current opened workspace(s) in browser?

### DISCUSSION THREAD

**Discussion by mister-teddy** [opensumi/core#1330] [Q&A] [answered]
In my component view I need to do this kind of logic:

```jsx
<div>
{openedWorkspace ? <><StartButton cwd={openedWorkspace} /><StopButton /></> : <NewProjectButton />}
</div>
```

But I'm struggling to get the opening workspace (I'm using the `ide-electron` template). Thank you for you support!

**Comment by erha19** [comment, marked as answer]
@nguyenhongphat0 maybe you can find the workspace path on the `window.metadata`, or you can use  `IWorkspaceService` token getting the WorkspaceService, like the samples here [use-service-by-di](https://opensumi.com/en/docs/develop/sample/use-service-by-di).

**Reply by mister-teddy** [reply]
Ah both `global.metadata.workspace` and 

```tsx
const workspaceService = useInjectable<IWorkspaceService>(IWorkspaceService);
const currentWorkspace = workspaceService.getWorkspaceRootUri(undefined);
```

did the trick. Thank you very much @erha19 🙏
